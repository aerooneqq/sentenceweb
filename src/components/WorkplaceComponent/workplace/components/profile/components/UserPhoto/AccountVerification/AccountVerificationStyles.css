#accountVerificationOutterContainer{ 
    width: 100%;
    height: auto;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    margin-top: 30px;
}

#accountVerificationCodeInput{ 
    width: 120px;
    height: 30px;

    box-sizing: border-box;

    padding-left: 3px;
    padding-right: 3px;

    border-width: 1px;
    border-style: solid;
    border-color: orangered;
    border-radius: 5px;

    font-family: Arial;
    font-size: 15px;
    color: gray;
}

    #accountVerificationCodeInput:focus{ 
        outline: none;
    }

#accountVerificationText{
    box-sizing: border-box;
    
    font-family: Arial; 
    color: black;
    font-size: 14px;

    padding-left: 10%;
    padding-right: 10%;

    margin-bottom: 10px;
}

#enterCodeContainer{ 
    display: flex; 
    flex-direction: row; 
    justify-content: center; 
    align-items: center;
}

#sendConfirmationCodeBtn{ 
    width: 70px;
    height: 30px;

    border-style: solid; 
    border-width: 1px;
    border-color: orange; 
    border-radius: 5px;

    background: white; 

    margin-left: 10px;

    font-size: 15px;
    color: orange;
}

    #sendConfirmationCodeBtn:focus{ 
        outline: none;
    }

    #sendConfirmationCodeBtn:hover{ 
        color: orangered;
        border-color: orangered;
    }

#verifiedAccountContainer { 
    width: 100%;
    height: auto; 

    display: flex;
    flex-direction: row;

    justify-content: center;
    align-items: center;
}

#verifiedAccountIcon { 
    width: 30px; 
    height: 30px;
}

#verifiedAccountText { 
    margin-left: 10px;

    font-size: 14px; 
    color: black;
}