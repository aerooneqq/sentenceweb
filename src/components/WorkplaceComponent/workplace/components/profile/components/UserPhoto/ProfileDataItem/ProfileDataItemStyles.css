.profileDataItem{
  width: 38px;
  height: 38px;

  background: orangered;

  border-width: 1px;
  border-color: orangered;
  border-style: solid;
  border-radius: 40px;

  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.profileDataItem:hover{
  background: orange;
  border-color: orange;
}

.profileDataItem:nth-child(1){
  transform: translate(0, -20px);
}
.profileDataItem:nth-child(2){
  transform: translate(-3px, 5px);
}
.profileDataItem:nth-child(3){
  transform: translate(0px, 15px);
}
.profileDataItem:nth-child(4){
  transform: translate(3px, 5px);
}
.profileDataItem:nth-child(5){
  transform: translate(0px, -20px);
}

.profileDataItemImg{
  width: 25px;
  height: 25px;

  background: transparent;

  border-style: none;
}
