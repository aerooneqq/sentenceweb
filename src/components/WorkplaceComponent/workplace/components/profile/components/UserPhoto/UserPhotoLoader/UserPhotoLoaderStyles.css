.userPhotoLoaderOutterCont { 
    width: 100%;
    height: 100%;

    display: flex;
    flex-direction: column;

    justify-content: center;
    align-items: center;
}

.userPhotoLoaderTopBar { 
    width: 90%;
    height: 40px;

    margin-top: 15px;

    background: #d0d0d0;

    border-style: none;
    border-radius: 5px;
}

.userPhotoLoaderPhotoCont { 
    margin-top: 20px;

    width: 200px; 
    height: 200px;

    border-radius: 150px;

    background: #d0d0d0;
}