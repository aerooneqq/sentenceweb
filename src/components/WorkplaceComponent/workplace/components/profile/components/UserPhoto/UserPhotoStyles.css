@media screen and (min-width: 400px){ 
  #photoBorder{
    margin-top: 30px;

    width: 200px;
    height: 200px;

    background: white;
  
    border-style: solid;
    border-width: 1px;
    border-radius: 250px;
    border-color: #F97646;

    box-sizing: content-box;

    padding: 0.1px;

    position: relative;
  }

  .userPhotoAndSettingsCont { 
    width: 100%;

    display: flex; 
    flex-direction: column;
    justify-content: center; 
    align-items: center;
  }

    #photoBorder:hover { 
      border-color: orangered;
    }

  .fillContainer { 
    flex-grow: 1;
  }

  .userPhotoTopSettings { 
    width: 100%;
    height: 50px;

    background: #e0e0e0;

    padding-left: 2.5%;
    padding-right: 2.5%;

    box-sizing: border-box;


    display: flex;
    flex-direction: row;  
    justify-content: baseline;
    align-items: center;
  }

  #userPhotoImg { 
    width: 200px;
    height: 200px;

    border-radius: 200px;
  }
  
  #photoContainer{  
    width: 300px; 
    
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  
  #photoBorder:hover #uploadNewUserPhotoContainer{
    visibility: visible;
    animation: uploadUserPhotoContAnimation 0.5s ease forwards;
  }
  
  @keyframes uploadUserPhotoContAnimation {
    from{
      opacity:  0;
    }
  
    to {
      opacity: 1;
    }
  }
  
  #profileDataItemsContainer{
    display: flex;
    flex-direction: row;
  }
  
  #uploadNewUserPhotoContainer{
    position: absolute;

    top: 0;
    left: 0;
  
    visibility: hidden;
    opacity: 0;
    width: 50px;
    height: 50px;
  
    border-style: none;
    border-radius: 50px;
    background: #DFDFDF;
  
    z-index: 2;
    margin-top: 120px;
    margin-left: 75px
  }
  
  #uploadNewUserPhotoContainer:hover {
    background: #E4E4E4;
  }
  
  #uploadNewUserPhotoImg{
    background: url("../../img/uploadNewUserPhoto.png");
  
    width: 24px;
    height: 24px;

    position: relative;
    left: 13px;
    top: 11px;
  }

  #uploadNewUserPhotoInput { 
    visibility: collapse;
  }
  
  #userPhotoloaderContainer{ 
    width: 100%;
    height: 100%;
  
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}

@media screen and (max-width: 400px){ 
  #photoBorder{
    width: 200px;
    height: 200px;
  
    background: white;
  
    border-style: solid;
    border-width: 0.7px;
    border-radius: 250px;
    border-color: orangered;
  
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .userPhotoAndSettingsCont { 
    width: 100%;
    
    display: flex; 
    flex-direction: column;
    justify-content: center; 
    align-items: center;
  }

  .fillContainer { 
    flex-grow: 1;
  }

  .userPhotoTopSettings { 
    width: 100%;
    height: 50px;

    padding-left: 2.5%;
    padding-right: 2.5%;

    box-sizing: border-box;

    background: #e0e0e0;

    display: flex;
    flex-direction: row;  
    justify-content: baseline;
    align-items: center;
  }
  
  #photoContainer{ 
    margin-top: 20px;

    display: flex;
    flex-direction: column;
    justify-content: center; 
    align-items: center;
  }
  
  #photoBorder:hover #uploadNewUserPhotoContainer{
    visibility: visible;
    animation: uploadUserPhotoContAnimation 0.5s ease forwards;
  }
  
  @keyframes uploadUserPhotoContAnimation {
    from{
      opacity:  0;
    }
  
    to {
      opacity: 1;
    }
  }
  
  #profileDataItemsContainer{
    display: flex;
    flex-direction: row;
  }
  
  #uploadNewUserPhotoContainer{
    position: relative;
  
    visibility: hidden;
    opacity: 0;
    width: 50px;
    height: 50px;
  
    border-style: none;
    border-radius: 50px;
    background: #DFDFDF;
  
    z-index: 2;
    margin-top: 140px;
  }
  
  #uploadNewUserPhotoContainer:hover {
    background: #E4E4E4;
  }
  
  #uploadNewUserPhotoImg{
    background: url("../../img/uploadNewUserPhoto.png");
  
    width: 24px;
    height: 24px;
  
    position: relative;
    left: 13px;
    top: 11px;
  }

  #uploadNewUserPhotoInput { 
    visibility: collapse;
  }
  
  #loaderContainer{ 
    width: 100%;
    height: 100%;
  
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}