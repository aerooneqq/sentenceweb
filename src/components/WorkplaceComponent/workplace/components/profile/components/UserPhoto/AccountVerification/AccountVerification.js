import React, {Component} from "react";

//Styles
import "./AccountVerificationStyles.css";

//Services
import CodesService from "../../../../../../../../services/CodesServices/CodesService";
import UserService from "../../../../../../../../services/UserServices/UserService";

//Icons
import verifiedAccountIcon from "./img/user_account_verified.png";

//Components
import Loader from "../../../../../../../loader/Loader";

//App messages
import {alertAppMessage} from "../../../../../../../ApplicationMessage/ApplicationMessageManager";


export default class AccountVerification extends Component { 
    constructor(props){ 
        super(props);

        this.codesService = new CodesService(localStorage.getItem("token"));
        this.userService = new UserService(localStorage.getItem("token"));

        this.state = { 
            inputValue: "",
            isLoading: true,
            isAccVerified: null
        };

        this.handleInputValueChange = this.handleInputValueChange.bind(this);
        this.handleSendCodeClick = this.handleSendCodeClick.bind(this);
        this.checkIfAccVerified = this.checkIfAccVerified.bind(this);
    }

    componentDidMount() { 
        this.checkIfAccVerified(); 
    } 

    checkIfAccVerified() { 
        this.userService.getPartialData(["isAccountVerified"])
            .then(res => { 
                this.setState({ 
                    isLoading: false,
                    isAccVerified: res.data.isAccountVerified
                });
            }).catch(er => { 
                if (er.reponse) { 
                    alertAppMessage(er.response.data, "error");
                }
            });
    }

    handleInputValueChange(event) {
        this.setState({ 
            inputValue: event.target.value
        });
    }

    handleSendCodeClick(event) { 
        let code = document.getElementById("accountVerificationCodeInput").value;

        this.setState({ 
            isLoading: true
        });

        this.codesService.activateAccount(code).then(res => { 
            if (res.status === 200) { 
                alertAppMessage("The account is verified.", "success");
                this.checkIfAccVerified();
            } 
        }).catch(er => { 
            if (er.response) { 
                alertAppMessage(er.response.data, "error");
            }
            else { 
                alertAppMessage("Unknown error did happen.", "error");
            }

            this.setState({
                isLoading: false
            });
        });
    }

    render() { 
        return (
            <div id = "accountVerificationOutterContainer">
                {this.state.isLoading === true ? 
                    <Loader message = "Getting verification data..." />
                    : 
                    this.state.isAccVerified === true ? 
                        <div id = "verifiedAccountContainer">
                            <img src = {verifiedAccountIcon} id = "verifiedAccountIcon"
                                alt = "Account verified"/>
                            <div id = "verifiedAccountText">
                                Your account is verified
                            </div>
                        </div>
                    :
                        <div>
                            <div id = "accountVerificationText">
                                Enter the code from the last e-mail. If you don't verify your account, it will be blocked.
                            </div>
                            <div id = "enterCodeContainer">
                                <input id = "accountVerificationCodeInput" type = "text"
                                    onChange = {this.handleInputValueChange} />
                                <button id = "sendConfirmationCodeBtn" onClick = {this.handleSendCodeClick}>
                                    Send
                                </button>
                            </div>
                        </div>
                }

            </div>
        );
    }
}