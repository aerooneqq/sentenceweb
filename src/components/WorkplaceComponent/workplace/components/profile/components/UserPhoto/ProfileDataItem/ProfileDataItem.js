import React, {Component} from "react"

//Styles
import "./ProfileDataItemStyles.css"

export default class ProfileDataItem extends Component{
  constructor(props){
    super(props)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick() { 
    this.props.data.changeUserData(this.props.data.currentUserData)
  }

  render(){
    return(
      <div className = "profileDataItem" onClick={this.handleClick}>
        <img src = {this.props.icon} className = "profileDataItemImg" alt = "" />
      </div>
    )
  }
}
