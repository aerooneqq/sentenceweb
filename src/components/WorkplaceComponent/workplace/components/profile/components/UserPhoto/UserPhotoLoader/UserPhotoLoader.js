import React, {Component} from "react";

//Styles
import "./UserPhotoLoaderStyles.css";


export default class UserPhotoLoader extends Component { 

    constructor(props) { 
        super(props);
    }

    render() { 
        return ( 
            <div className = "userPhotoLoaderOutterCont">

                <div className = "userPhotoLoaderTopBar"/>
                <div className = "userPhotoLoaderPhotoCont" />

            </div>
        )
    }
}