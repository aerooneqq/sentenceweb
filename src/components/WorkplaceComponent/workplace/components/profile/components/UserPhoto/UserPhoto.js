import React, {Component} from "react";

//Styles
import "./UserPhotoStyles.css";

//Icons
import careerIcon from "./img/career_profile_data_item.svg";
import authorizationIcon from "./img/authorization_profile_data_item.svg";
import friendsIcon from "./img/friends_profile_data_item.svg"; 
import locationIcon from "./img/location_profile_data_item.svg";
import nameIcon from "./img/name_profile_data_item.svg";

//Services
import { alertAppMessage } from "../../../../../../ApplicationMessage/ApplicationMessageManager";
import ResponseService from "../../../../../../../services/ResponseService/ReponseService";
import UserPhotoService from "../../../../../../../services/UserPhotoService/UserPhotoService";

//Components
import Loader from "../../../../../../loader/Loader";
import UserPhotoLoader from "./UserPhotoLoader/UserPhotoLoader";
import ProfileDataItem from "./ProfileDataItem/ProfileDataItem";
import SignOutComponent from "./SignOut/SignOut";
import DeleteAccountComponent from "./DeleteAccount/DeleteAccount";
import AccountVerification from "./AccountVerification/AccountVerification";

/**
 * PROPS LIST:
 * 1) changeUserData - the function to change the currently displayed data.
 * 2) signOut  - the function to sign out from the system
 */
export default class UserPhoto extends Component {
  _icons = [careerIcon, authorizationIcon, nameIcon, locationIcon, friendsIcon];
  _toolTips = ["Career", "Authentication", "Name", "Location", "Subscribers and subscriptions"];

  constructor(props) {
    super(props);

    this.userPhotoService = new UserPhotoService(localStorage.getItem("token"));
    this.responseService = new ResponseService();

    let profileDataItems = [];

    for (let i = 0; i < 5; i++){ 
      let props = { 
        imgURL: careerIcon,
        currentUserData: i,
        changeUserData: this.props.changeUserData,

      }

      profileDataItems.push(<ProfileDataItem data = {props} 
                                             icon = {this._icons[i]}
                                             toolTip = {this._toolTips[i]}/>);
    }

    this.state = { 
      profileDataItems: profileDataItems,
      isPhotoUpdating: true,
      isFirstLoad: true,
      userPhoto: null
    }

    this.updateUserPhoto = this.updateUserPhoto.bind(this);
    this.uploadUserPhoto = this.uploadUserPhoto.bind(this);
  }

  componentDidMount() { 
    this.uploadUserPhoto();
  }

  uploadUserPhoto() { 
    this.userPhotoService.getUserPhoto().then(res => { 
      this.setState({ 
        isPhotoUpdating: false,
        isFirstLoad: false,
        userPhoto: res.data.photo
      });

      document.getElementById("userPhotoImg").src = `data:image/png;base64, ${this.state.userPhoto}`
    }).catch(er => { 
      this.setState({ 
        isPhotoUpdating: false,
        isFirstLoad: false,
      });
      this.responseService.alertErrorMessage(er, "Error occured while getting photo.");
    });
  }

  updateUserPhoto() { 
    let input = document.createElement("input");
    input.type = "file";
    input.accept = "image/jpeg,image/png,image/gif";

    input.onchange = e => { 
      let file = e.target.files[0];

      let reader = new FileReader();
      reader.readAsArrayBuffer(file);

      reader.onload = e => { 
        let array = new Uint8Array(e.target.result);
        
        let byteArray = []

        for (let i = 0; i < array.length; i++) { 
          byteArray.push(array[i]);
        }

        this.setState({ 
          isPhotoUpdating: true
        });

        this.userPhotoService.updateUserPhoto(byteArray).then(() => { 
          alertAppMessage("The profile photo was updated!", "success");
          this.setState({ 
            isPhotoUpdating: false
          });
        }).catch(er => { 
          this.responseService.alertErrorMessage(er, "The error occured when updating the photo");
          this.setState({ 
            isPhotoUpdating: false
          });
        });
      }
    }

    input.click();
  }

  render() {
      return (
        <div id="photoContainer">
          
          {this.state.isFirstLoad === true ? <UserPhotoLoader /> :
              <div className = "userPhotoAndSettingsCont">
                <div className = "userPhotoTopSettings">
                  <SignOutComponent signOut = {this.props.signOut}/>
                  <div className = "fillContainer" />
                  <DeleteAccountComponent />
                </div>
      
                <div id="photoBorder">
                  {this.state.isPhotoUpdating === true ? <Loader message = "Updating the photo..."/> : 
                  <div>
                    <img id = "userPhotoImg" alt = "Profile" />
                    <div id ="uploadNewUserPhotoContainer" onClick = {this.updateUserPhoto}>
                      <div id="uploadNewUserPhotoImg"></div>
                    </div> 
                  </div>}
      
                </div>
                <div id="profileDataItemsContainer">
                  {this.state.profileDataItems}
                </div>
              </div>
          }
          <AccountVerification />
        </div>
      )
  }
}
