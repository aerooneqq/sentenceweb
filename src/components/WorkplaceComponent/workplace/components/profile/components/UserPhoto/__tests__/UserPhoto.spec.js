import React from 'react'
import { shallow, render } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import UserPhoto from "../UserPhoto";

import Loader from "../../../../../../../loader/Loader";
import UserPhotoLoader from "../UserPhotoLoader/UserPhotoLoader";
import ProfileDataItem from "../ProfileDataItem/ProfileDataItem";
import SignOutComponent from "../SignOut/SignOut";
import DeleteAccountComponent from "../DeleteAccount/DeleteAccount";
import AccountVerification from "../AccountVerification/AccountVerification";

describe("UserPhoto", () => { 
    it("should render corectly", () => { 
        const userPhoto = shallow(<UserPhoto />);

        expect(shallowToJson(userPhoto)).toMatchSnapshot();
    });

    it("should have proper initial state", () => { 
        const userPhoto = shallow(<UserPhoto />);

        const rightState = { 
            isPhotoUpdating: true,
            isFirstLoad: true,
            userPhoto: null
        }

        for (let key in rightState) { 
            expect(userPhoto.state(key)).toBe(rightState[key]);
        }
    });

    it("should have UserPhotoLoader when intial render", () => { 
        const userPhoto = shallow(<UserPhoto />);

        expect(userPhoto.find(UserPhotoLoader).exists()).toBe(true);
        expect(userPhoto.find(AccountVerification).exists()).toBe(true);
    });

    it("should have all utility components when loaded", () => { 
        const userPhoto = shallow(<UserPhoto />);

        userPhoto.setState({
            isFirstLoad: false,
            isPhotoUpdating: false
        });

        expect(userPhoto.find(SignOutComponent).exists()).toBe(true);
        expect(userPhoto.find(AccountVerification).exists()).toBe(true);
        expect(userPhoto.find(DeleteAccountComponent).exists()).toBe(true);
    });
});