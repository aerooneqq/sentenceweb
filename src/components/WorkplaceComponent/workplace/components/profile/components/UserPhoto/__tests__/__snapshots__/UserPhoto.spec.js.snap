// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPhoto should render corectly 1`] = `
<div
  id="photoContainer"
>
  <UserPhotoLoader />
  <AccountVerification />
</div>
`;
